### PR TITLE
Feat: NodeJS runtime

### PR DESCRIPTION
**Pull request by Meldiron**
Opened questions:

- I struggled to figure out how to actually run example code, and I still didn't figure it out. I understand there needs to be a build process before, but I could not get it to work. I think we should focus on developer experience more, and prepare simple commands for developers. We should hide build complexity behind some simple docker run command, similar to how Appwrite does installation, or composer update.

### CODE REVIEW

**Comment by eldadfux** [comment]
> Opened questions:
> 
> * I struggled to figure out how to actually run example code, and I still didn't figure it out. I understand there needs to be a build process before, but I could not get it to work. I think we should focus on developer experience more, and prepare simple commands for developers. We should hide build complexity behind some simple docker run command, similar to how Appwrite does installation, or composer update.

Maybe a simple docker-compose file?

@Meldiron what about the tests? can we gradually move them here as well?

**Comment by Meldiron** [comment]
> > Opened questions:
> > 
> > * I struggled to figure out how to actually run example code, and I still didn't figure it out. I understand there needs to be a build process before, but I could not get it to work. I think we should focus on developer experience more, and prepare simple commands for developers. We should hide build complexity behind some simple docker run command, similar to how Appwrite does installation, or composer update.
> 
> Maybe a simple docker-compose file?
> 
> @Meldiron what about the tests? can we gradually move them here as well?

@eldadfux Should we merge tests with an example script? If an example script can execute and returns the expected response, I would say the runtime can pass. Current test does not test building nor library usage at all.

**Comment by eldadfux** [comment]
I think we can skip building for now and test it on Appwrite itself.

File: runtimes/build-local.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+echo 'Starting local build...'
+
+echo 'Node 17...'
+docker build -t open-runtimes/node:17.0 ./runtimes/node-17.0
```
Comment (christyjacob4) [inline runtimes/build-local.sh:4]: I think these paths need to be fixed. Keep in mind the folder structure

Comment (christyjacob4) [inline runtimes/build-local.sh:4]:
```suggestion
docker build -t open-runtimes/node:17.0 ./node-17.0
```

File: runtimes/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo 'Starting build...'
+
+echo 'Node 17...'
+docker buildx build --platform linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64,linux/ppc64le -t open-runtimes/node:17.0 ./runtimes/node-17.0/ --push
```
Comment (christyjacob4) [inline runtimes/build.sh:4]:
```suggestion
docker buildx build --platform linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64,linux/ppc64le -t open-runtimes/node:17.0 ./node-17.0/ --push
```

File: runtimes/build-local.sh (hunk 1 above)
Comment (Meldiron) [reply runtimes/build-local.sh]: Resolved by other suggestions

File: runtimes/node-17.0/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+git clone https://github.com/open-runtimes/open-runtimes.git
+```
+
+Enter the node runtime folder:
+
+```bash
+cd open-runtimes/runtimes/node-17.0
+```
+
+Run the example script:
```
Comment (christyjacob4) [inline runtimes/node-17.0/README.md:21]:
```suggestion
Run the included example cloud function:
```

File: runtimes/node-17.0/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+```bash
+docker-compose up -d
+```
+
+You can now send `POST` request to `http://localhost:3000`. Make sure you have header `x-internal-challenge: example1234`, and JSON body `{ "path": "/usr/code", "file": "index.js" }`.
+
+If you make changes to example script, restart to apply changes with `docker-compose restart`.
```
Comment (christyjacob4) [inline runtimes/node-17.0/README.md:29]:
```suggestion
You can also make changes to the example code and apply the changes with the `docker-compose restart` command.
```

File: runtimes/node-17.0/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# Node Runtime 17.0
+
+This is the Open Runtime that builds and runs NodeJS code based on a `node:17-alpine` base image. 
+
+The runtime itself uses [Micro](https://github.com/vercel/micro) as the Web Server to process the execution requests.
```
Comment (christyjacob4) [inline runtimes/node-17.0/README.md:5]: The readme looks good but I think you should include a few more points in this section about the folder structure. 
You can briefly mention the purpose of the files and folders here like

```
example/
launch.sh
server.js
```

Or any other files that are not obvious.

File: runtimes/node-17.0/README.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+docker-compose up -d
+```
+
+You can now send `POST` request to `http://localhost:3000`. Make sure you have header `x-internal-challenge: example1234`, and JSON body `{ "path": "/usr/code", "file": "index.js" }`.
+
+If you make changes to example script, restart to apply changes with `docker-compose restart`.
+
+# Notes
+
+Functions for this runtime must be exported directly through the `module.exports` object.
```
Comment (christyjacob4) [inline runtimes/node-17.0/README.md:33]:
```suggestion
When writing functions for this runtime , ensure they are directly through the `module.exports` object.
```

**Review by christyjacob4 — APPROVED** [review]
Please address the comments

**Pull request merged**